modules/core/src/utils/math-utils.ts

```diff
@@ -132,3 +132,19 @@ export function toDoublePrecisionArray(
 
   return scratchArray.subarray(0, count * size * 2);
 }
+
+type LayerBounds = [number[], number[]] | null;
+export function mergeBounds(bounds1: LayerBounds, bounds2: LayerBounds): LayerBounds {
```

Comment:
Adding null to a clearly defined type doesn't really make sense. Better add that in the function declaration.
```suggestion
type LayerBounds = [number[], number[]];
export function mergeBounds(bounds1: LayerBounds | null, bounds2: LayerBounds | null): LayerBounds | null {
}
```
Nit: While it is sometimes tempting to push null checks into functions (pushing such details out of the main logic), it does erode the clarity of helpers. E.g. do we need to support both of these arguments being null or could we restrict that to one of them and in that case drop the null from the return type?